.github/workflows/ci.yml

```diff
@@ -83,13 +83,79 @@ jobs:
 
       - name: Run Django Tests with coverage
         run: docker-compose run django coverage run -m pytest -vvvv -rsa -m "not local"
-
       - name: Generate coverage XML
         run: docker-compose run django coverage xml
 
       - name: Run mypy
         run: docker-compose run django mypy ds_judgements_public_ui judgments
 
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+
+      - name: Upload coverage to CodeClimate
+        uses: paambaati/codeclimate-action@v5.0.0
+        env:
+          CC_TEST_REPORTER_ID: ${{secrets.CC_TEST_REPORTER_ID}}
+
+      - name: Tear down the Stack
+        run: docker-compose down
+
+  e2e-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code Repository
+        uses: actions/checkout@v4
+
+      - name: Copy .env.example file
+        uses: canastro/copy-file-action@master
+        with:
+          source: ".env.example"
+          target: ".env"
+
+      - name: Create docker network as used in dev
+        run: docker network create caselaw
+
+      - name: Build the Stack
+        run: docker-compose build postgres django
+
+      - name: Checkout Marklogic repo
+        uses: actions/checkout@v4
+        with:
+          repository: "nationalarchives/ds-caselaw-marklogic"
+          path: "./marklogic"
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+
+      - name: Install requests
+        run: "pip install requests"
+
+      - name: Run marklogic docker container
+        run: docker-compose up -d
+        working-directory: "./marklogic"
+
+      - name: Deploy marklogic configuration
+        run: sleep 30 && gradle mlDeploy
```

Comment:
(as per [Slack convo](https://dxw.slack.com/archives/C04FQ3GFGUQ/p1719300443189719)) we should be able to ditch this `sleep` by adding a [`HEALTHCHECK`](https://docs.docker.com/reference/dockerfile/#healthcheck) to the Dockerfile pointing to the [MarkLogic healthcheck](https://docs.marklogic.com/11.0/guide/release-notes/en/new-features-in-marklogic-11/enhanced-health-check-monitoring.html), and then sticking `--wait` on the above `docker-compose up`.

This should cause the entire process to pause until everything is reporting green.